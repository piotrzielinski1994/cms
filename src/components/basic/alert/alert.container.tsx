import { useTranslations } from 'next-intl';
import { ComponentProps } from 'react';
import { Alert } from './alert';

const AlertContainer = (props: ComponentProps<typeof Alert>) => {
  const t = useTranslations('frontend');
  return <Alert t={{ close: t('close') }} {...props} />;
};

export { AlertContainer };
