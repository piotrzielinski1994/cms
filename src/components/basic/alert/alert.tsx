import { cn } from '@/utils/tailwind';
import { AlertTriangle, CheckCircle, Info, XCircle } from 'lucide-react';
import { JSX, PropsWithChildren } from 'react';

type AlertType = 'success' | 'info' | 'warn' | 'error';
type AlertProps = PropsWithChildren & {
  type?: AlertType;
};

const Alert = ({ type = 'info', children }: AlertProps) => {
  return (
    <div role="alert" className={cn('p-2', 'flex gap-2 flex-wrap', colorMap[type])}>
      {iconMap[type]}
      <div>{children}</div>
    </div>
  );
};

const iconMap = {
  success: <CheckCircle className="w-[1lh] h-[1lh]" />,
  info: <Info className="w-[1lh] h-[1lh]" />,
  warn: <AlertTriangle className="w-[1lh] h-[1lh]" />,
  error: <XCircle className="w-[1lh] h-[1lh]" />,
} satisfies Record<AlertType, JSX.Element>;

const colorMap = {
  success: 'bg-green-500 text-green-950',
  info: 'bg-blue-500 text-blue-950',
  warn: 'bg-orange-500 text-orange-950',
  error: 'bg-red-500 text-red-950',
} satisfies Record<AlertType, string>;

export { Alert };
