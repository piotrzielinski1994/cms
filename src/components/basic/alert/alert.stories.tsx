import { DEFAULT_VALUE, getFallback } from '@/config/storybook/utils';
import type { Meta, StoryObj } from '@storybook/react';
import { useTranslations } from 'next-intl';
import { type ComponentProps } from 'react';
import { Alert as AlertComponent } from './alert';

type Args = ComponentProps<typeof AlertComponent>;

const meta = {
  component: AlertComponent,
  title: 'Components/Basic/Alert',
  argTypes: {
    type: {
      control: 'select',
      options: ['info', 'warn', 'error'] satisfies Array<Args['type']>,
    },
    children: { control: 'text' },
  },
  args: {
    type: 'info',
    children: DEFAULT_VALUE,
  },
} satisfies Meta<Args>;

const Render = ({ children, ...args }: Args) => {
  const t = useTranslations('storybook.basic.alert');
  return (
    <div className="grid gap-4">
      <AlertComponent {...args}>{getFallback(children, t('info'))}</AlertComponent>
      <AlertComponent type="success">{t('success')}</AlertComponent>
      <AlertComponent type="warn">{t('warn')}</AlertComponent>
      <AlertComponent type="error">{t('error')}</AlertComponent>
    </div>
  );
};

const Alert: StoryObj<typeof AlertComponent> = { render: Render };

export { Alert };
export default meta;
