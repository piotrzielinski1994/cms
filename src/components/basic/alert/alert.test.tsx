import { render } from '@testing-library/react';
import { ComponentProps } from 'react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import { Alert } from './alert';

describe('Alert', () => {
  const defaultItems = [
    { heading: 'Heading 1', content: 'Content 1' },
    { heading: 'Heading 2', content: 'Content 2' },
  ] satisfies ComponentProps<typeof Alert>['items'];

  it('should have no accessibility violations', async () => {
    const { container } = render(<Alert items={defaultItems} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(<Alert items={defaultItems} />);
    expect(container).toMatchSnapshot();
  });
});
