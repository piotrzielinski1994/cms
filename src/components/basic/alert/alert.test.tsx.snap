// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > should match the snapshot 1`] = `
<div>
  <div
    class="p-2 flex gap-2 flex-wrap bg-blue-500 text-blue-950"
    role="alert"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-info w-[1lh] h-[1lh]"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
    <div />
  </div>
</div>
`;
